packages/sdk/src/contracts/OperatorRegistry.ts

```diff
@@ -55,4 +69,32 @@ export class OperatorRegistry {
         this.logger.debug(`Found ${peerDescriptors.length} network entrypoints, picked ${picked.length}`, { picked })
         return picked
     }
+
+    async findOperatorsOnStream(streamId: StreamID, maxQueryResults: number, maxHeartbeatAgeHours: number): Promise<NetworkPeerDescriptor[]> {
+        const query: GraphQLQuery = { 
+            query: `{
+                stream(id: "${streamId}") {
+                    sponsorships(where: { isRunning: true }) {
+                        stakes(first: ${maxQueryResults}, orderBy: updateTimestamp, orderDirection: desc) {
+                            operator (
+                                where: {
+                                    latestHeartbeatMetadata_contains: "\\"tls\\":true", 
+                                    latestHeartbeatTimestamp_gt: "${Math.floor(Date.now() / 1000) - (maxHeartbeatAgeHours * 60 * 60)}"
+                                }
+                            ) {
+                                id
+                                latestHeartbeatMetadata
+                            }
+                        }
+                    }
+                }
+            }`
+        }
+        const operatorMetadatas = await this.theGraphClient.queryEntity<StreamOperators>(query)
+        const peerDescriptors: NetworkPeerDescriptor[] = operatorMetadatas.stream.sponsorships
+            .flatMap((sponsorship: Sponsorship) => sponsorship.stakes
+                .map((stake: { operator: OperatorMetadata }) => JSON.parse(stake.operator.latestHeartbeatMetadata)))
```

Comment:
Naming is misleading here
```suggestion
        const results = await this.theGraphClient.queryEntity<StreamOperators>(query)
        const peerDescriptors: NetworkPeerDescriptor[] = results.stream.sponsorships
            .flatMap((sponsorship: Sponsorship) => sponsorship.stakes
                .map((stake: { operator: OperatorMetadata }) => JSON.parse(stake.operator.latestHeartbeatMetadata)))
```